### PR TITLE
test(engine/dm): add engine dm chaos test

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref #5504 

### What is changed and how it works?
- use simulator as upstream workload
- add chaos test for engine dm
- upgrade chaos mesh to v2

#### Tests 

 - chaos test

### Release note 

```release-note
None
```

### CODE REVIEW

File: dm/simulator/config/config.go [hunk 1]
```diff
@@ -29,3 +36,32 @@ type ColumnDefinition struct {
...
+		buf.WriteString(dbutil.ColumnName(col.ColumnName))
+		buf.WriteByte(' ')
+		buf.WriteString(col.DataType)
+		if col.DataLen > 0 {
+			buf.WriteByte('(')
+			buf.WriteString(strconv.Itoa(col.DataLen))
+			buf.WriteByte(')')
+		}
+	}
+	for _, ukColName := range t.UniqueKeyColumnNames {
```
Comment (lance6716) [inline dm/simulator/config/config.go:58]: not sure `UniqueKeyColumnNames` represents multiple columns in one UK or multiple UK which has one column

**Review by lance6716 — COMMENTED** [review]
will review the big file later

File: dm/simulator/config/config.go (hunk 1 above)
Comment (GMHDBJD) [reply dm/simulator/config/config.go]: It's multiple columns in one UK, fix it.

**Review by lance6716 — APPROVED** [review]

File: .github/workflows/dataflow_engine_chaos.yaml [hunk 2]
```diff
@@ -9,6 +9,9 @@ on:
         description: 'Which PR do you want to trigger (use PR number, such as 6127)'
         required: true
         default: ''
+  pull_request:
+    branches:
+      - master
```
[the remark this answers is not in the document]
Comment (GMHDBJD) [reply .github/workflows/dataflow_engine_chaos.yaml:14]: Yes

File: .github/workflows/dm_chaos.yaml [hunk 3]
```diff
@@ -9,6 +9,9 @@ on:
         description: 'Which PR do you want to trigger'
         required: true
         default: ''
+  pull_request:
+    branches:
+      - master
```
Comment (amyangfei) [inline .github/workflows/dm_chaos.yaml:14]: ditto

**Review by amyangfei — APPROVED** [review]

**Pull request merged**